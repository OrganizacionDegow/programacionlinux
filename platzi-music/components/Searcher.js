import React from 'react';
import styled from 'styled-components';

const DefaultStyles = `
  outline: 0;
  border: none;
  border-radius: 4px;
`;

const Form = styled.form`
  display: flex;
  font-weight: bold;
  font-family: 'Quicksand';
  @media (max-width: 1024px) {
    flex-direction: column;
  }
`;

const InputSearcher = styled.input`
  flex: 1;
  margin-right: 1rem;
  padding: 0.5rem 1rem;
  font-size: 1rem;
  ${DefaultStyles}
  @media (max-width: 1024px) {
    margin-right: 0;
  }
`;

const Button = styled.button`
  ${DefaultStyles}
  background: #ea83ee;
  padding: 0.5rem 46px;
  text-transform: uppercase;
  font-weight: bold;
  color: white;
  box-shadow: 0 0.25rem 0.5rem 0 rgba(0, 0, 0, 0.3);
  font-size: 1rem;
  font-family: 'Quicksand';
  @media (max-width: 1024px) {
    margin-top: 1rem;
  }
`;

// es invocado desde Hero
function Searcher(props) {
  return(
    <Form onSubmit={props.onSubmit}>
      <InputSearcher
        name="buscar"
        type="text"
        placeholder="Busca por canción, artista o álbum"
      />
      <Button>Buscar</Button>
    </Form>
  );
}

export default Searcher;
