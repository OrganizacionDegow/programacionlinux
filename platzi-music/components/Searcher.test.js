import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import Searcher from './Searcher';

function handleSubmit() {}

describe('Test Searcher component', () => {
  it('should render', () => {
    const div = document.createElement('div');
    render(
      <Searcher onSubmit={handleSubmit} />,
      div,
    );
  });

  it('should render the expected HTML', () => {
    const component = renderer.create(
      <Searcher onSubmit={handleSubmit} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
