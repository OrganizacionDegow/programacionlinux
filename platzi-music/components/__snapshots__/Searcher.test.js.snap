// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Searcher component should render the expected HTML 1`] = `
<form
  className="bdvyEE"
  onSubmit={[Function]}
>
  <input
    className="WqvUF"
    name="buscar"
    placeholder="Busca por canción, artista o álbum"
    type="text"
  />
  <button
    className="eJPHrw"
  >
    Buscar
  </button>
</form>
`;
